### PR TITLE
Added NOXP reader for IPHEx

### PR DESCRIPTION
**Pull request by tjlang**
For IPHEx, NOXP radar produced netCDF3-formatted data that is currently
not readable by either Radx or Py-ART. I wrote a sub-component of the
pyart.aux_io module to ingest these data into the Radar object. A
modified version of the read_d3r_gcpex_nc function template was used
for this.

caveats:
1. No formal unit tests yet as I need NOAA to provide a properly
formatted test file. Script tested at the unit level using real data
and an in-house Jupyter notebook.
2. Untested on RHI data as these files have not yet been provided by
NOAA.

screen shot 2015-07-15 at 9 46 35 am

### CODE REVIEW

**Comment by jjhelmus** [comment]
@tjlang Looks like a good addition to Py-ART and should help anyone who runs into these files. I'm currently traveling but will try to have a more in depth look at this by the end of the week.

File: pyart/aux_io/noxp_iphex_nc.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                       file_field_names=False, exclude_fields=None, **kwargs):
+    """
+    Read a NOXP IPHEX netCDF file.
+
+    Parameters
+    ----------
+    filename : str
+        Name of the netCDF file to read.
+    field_names : dict, optional
+        Dictionary mapping ODIM_H5 field names to radar field names. If a
```
Comment (jjhelmus) [inline pyart/aux_io/noxp_iphex_nc.py]: "netCDF field names"? not "ODIM_H5 field names"

File: pyart/aux_io/noxp_iphex_nc.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        A value of None, the default, will use the mapping defined in the
+        Py-ART configuration file.
+    additional_metadata : dict of dicts, optional
+        Dictionary of dictionaries to retrieve metadata from during this read.
+        This metadata is not used during any successive file reads unless
+        explicitly included.  A value of None, the default, will not
+        introduct any addition metadata and the file specific or default
+        metadata as specified by the Py-ART configuration file will be used.
+    file_field_names : bool, optional
+        True to use the MDV data type names for the field names. If this
```
Comment (jjhelmus) [inline pyart/aux_io/noxp_iphex_nc.py]: "netCDF variable names" not "MDV data type nameS"

**Comment by jjhelmus** [comment]
@tjlang Looks like a good addition to Py-ART.  I made a few suggestions to clean up the docstring.  Also a line should be added to the [doc/source/dev_reference/aux_io.rst](https://raw.githubusercontent.com/ARM-DOE/pyart/master/doc/source/dev_reference/aux_io.rst) file so that the module appears in the Developer Reference Manual.  With those fixes this should be good to merge.

**Comment by tjlang** [comment]
OK, I will work on this after I get back from vacation.

**Comment by tjlang** [comment]
Made the requested updates ...

**Comment by jjhelmus** [comment]
Looks good, thanks again Tim.

**Pull request merged**